### PR TITLE
add copy invoiceAmount button

### PR DESCRIPTION
**Pull request by easter-monolith**
@tiero please review

### CODE REVIEW

File: components/modals/invoice.tsx [hunk 1]
```diff
@@ -75,17 +76,49 @@ const InvoiceModal = ({ contract, handler }: InvoiceModalProps) => {
...
+
   const pn = (n: number, precision: number) =>
-    prettyNumber(fromSatoshis(n, precision), 0, precision)
+    prettyNumber(fromSatoshis(n, precision), precision, precision)
+
+  const copyInvoiceAmount = () => {
+    navigator.clipboard.writeText(pn(invoiceAmount, collateral.precision)).then(
+      () => {
+        const target = document.getElementById('copyInfo')
+        console.log('target', target)
```
Comment (tiero) [inline components/modals/invoice.tsx:87]: remove console.log

File: components/modals/invoice.tsx [hunk 2]
```diff
@@ -75,17 +76,49 @@ const InvoiceModal = ({ contract, handler }: InvoiceModalProps) => {
...
   const Separator = () => <p className="mx-3"> &ndash; </p>
+
   const pn = (n: number, precision: number) =>
-    prettyNumber(fromSatoshis(n, precision), 0, precision)
+    prettyNumber(fromSatoshis(n, precision), precision, precision)
+
+  const copyInvoiceAmount = () => {
+    navigator.clipboard.writeText(pn(invoiceAmount, collateral.precision)).then(
+      () => {
+        const target = document.getElementById('copyInfo')
```
Comment (tiero) [inline components/modals/invoice.tsx:86]: do we really need to access DOM directly? can;t we use `invoiceAmount`

File: components/modals/invoice.tsx [hunk 3]
```diff
@@ -75,17 +76,49 @@ const InvoiceModal = ({ contract, handler }: InvoiceModalProps) => {
...
+        console.log('target', target)
+        if (!target) return false
+        target.style.visibility = 'visible'
```
Comment (tiero) [inline components/modals/invoice.tsx:89]: Maybe best to create a dedicated component ie. "CopyButton" that has a internal state `buttonText` where the text go from `Copy` to `Copied` and with a sleep function back to `Copy`?

https://github.com/tiero/sats-starter/blob/4423c4c5df4ef5c80b7deb64ce05911447b7b8d6/components/payment.tsx#L43-L51

Comment (easter-monolith) [reply components/modals/invoice.tsx]: You're right, much cleaner with [fb71cf2](https://github.com/fuji-money/app/pull/147/commits/fb71cf2aa43c9f0b2ca5d6bd5533150c079c500b)

**Pull request merged**